common/txmgr/address_state.go

```diff
@@ -465,6 +465,7 @@ func (as *addressState[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) _findT
 				for i := 0; i < len(tx.TxAttempts); i++ {
 					txAttempt := tx.TxAttempts[i]
 					if txAttemptFilter(&txAttempt) {
+						txAttempt.Tx = *tx
```

Comment:
Does a `TxAttempt` really need to contain the entire Tx, or does it just need the ID? I don't think we should be getting a Tx from an attempt (only the other way around to get the list of attempts from a Tx). In the worst case I think this should be a pointer to a Tx so we're not copying Tx onto each attempt multiple times.